### PR TITLE
Improve dependency structure

### PR DESCRIPTION
**Pull request by mthmulders**
Following up on #74, here's a suggestion how to improve the dependencies between MockRunners modules.

Please let me know what you think!

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -202,6 +202,28 @@
...
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.mockrunner</groupId>
+                <artifactId>mockrunner-servlet</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.mockrunner</groupId>
+                <artifactId>mockrunner-core</artifactId>
```
Comment (cemartins) [inline pom.xml]: mockrunner-core is defined twice in dependency management. That won't work. We must have only one, and then perhaps this specific type:xml, classifier:features can be overrides in the mock runner-servlet module?

File: mockrunner-servlet/pom.xml [hunk 2]
```diff
@@ -22,20 +22,14 @@
...
 			<artifactId>mockrunner-core</artifactId>
-			<version>2.0.4-SNAPSHOT</version>
-			<scope>provided</scope>
 		</dependency>
 		<dependency>
 			<groupId>com.mockrunner</groupId>
 			<artifactId>mockrunner-core</artifactId>
-			<version>2.0.4-SNAPSHOT</version>
-			<type>xml</type>
-			<classifier>features</classifier>
```
Comment (cemartins) [inline mockrunner-servlet/pom.xml]: Maybe leave these two lines, and in the parent pom define the version.

File: pom.xml [hunk 3]
```diff
@@ -202,6 +202,28 @@
                 <artifactId>struts-taglib</artifactId>
                 <version>1.3.10</version>
             </dependency>
+            <dependency>
+                <groupId>com.mockrunner</groupId>
+                <artifactId>mockrunner-core</artifactId>
+                <version>${project.version}</version>
```
Comment (cemartins) [inline pom.xml]: Good idea. I like it!

File: pom.xml (hunk 1 above)
Comment (mthmulders) [reply pom.xml]: You're right that it won't work, but for a different reason.

The first occurrence is the MockRunner Core (JAR), the second one should be the Features classifier of type XML. But you can't "override" the classifier or the type in the child module - everything except version and scope must be equal.

So I've updated the child module to also mention we're looking for the Features classifier of type XML, and now it should resolve to the correct version as declared in the parent Dependency Management.

File: pom.xml (hunk 3 above)
Comment (mthmulders) [reply pom.xml]: Thanks :-)

File: mockrunner-servlet/pom.xml (hunk 2 above)
Comment (mthmulders) [reply mockrunner-servlet/pom.xml]: See above

**Comment by mthmulders** [comment]
> I noticed you added yourself as contributor ;)
> ... are you sure you don't want to be part of the team of maintainers?

Yes, positively sure, but thanks for the offer. Thanks for the offer, though.

**Comment by cemartins** [comment]
You’re welcome. And thanks again for the contribution.

**Pull request merged**